### DISCUSSION TITLE
Way to hide a Subcategory

### DISCUSSION THREAD

**Discussion by hsanand61564** [iTwin/itwinjs-core#486] [Q&A] [answered]
Hi

I am trying to hide a subcategory in my imodel, but not getting success into the same. Below is the snippet code I am using for the same.

![image](https://user-images.githubusercontent.com/60655118/102881875-85b39a00-4473-11eb-8d03-7cbf19632d65.png)

Any help in this regard would be really appreciable.

**Comment by pmconne** [comment, marked as answer]
[The docs for DisplayStyleState.overrideSubcategory](https://www.itwinjs.org/reference/imodeljs-frontend/views/displaystylestate/overridesubcategory/?term=overridesubcategory) have this note:
>If this style is associated with a ViewState attached to a Viewport, use Viewport.overrideSubCategory to ensure the changes are promptly visible on the screen.

I happen to be working on a PR right now that would eliminate the need to call `Viewport.overrideSubCategory` by automatically synchronizing the viewport's state when the display style's state changes, but for now you must use the Viewport API.

**Reply by pmconne** [reply]
The PR is #491.
